src/test/rustdoc/impl-on-foreign.rs

```diff
@@ -0,0 +1,26 @@
+#![crate_name = "foo"]
+
+use std::convert::AsRef;
+pub struct Gus;
+
+// @has 'foo/struct.Gus.html'
+// @has - '//h3[@class="impl"]/code' 'impl AsRef<str> for Gus'
+impl AsRef<str> for Gus {
+    fn as_ref(&self) -> &str {
+        todo!()
+    }
+}
+
+// @has - '//h3[@class="impl"]/code' 'impl AsRef<Gus> for str'
+impl AsRef<Gus> for str {
+    fn as_ref(&self) -> &Gus {
+        todo!()
+    }
+}
+
+// @has - '//h3[@class="impl"]/code' 'impl AsRef<Gus> for String'
+impl AsRef<Gus> for String {
```

Comment:
is std considered special in anyway in rustdoc? or can we just consider it any old external crate?

Reply:
No, it's considered as an external crate.